RELEASE.md

```diff
@@ -1,5 +1,28 @@
+# Release
+
 To release a new version of Spyder you need to follow these steps:
 
+## Localization updates
+
+* Install [gettext-helper](https://github.com/spyder-ide/gettext-helpers) from source.
+
+* Update `*.pot` and `*.po` files to update translation strings using
+  * Run `spyder-gettext scane spyder` to update localization files.
+  * Create and merge a new PR with these updated files
+  * Request translators to update the translation strings on https://crowdin.com/project/spyder
+
+* Close the automatic translation bot PR `translate/<branch-name>` and delete the branch to recreate the branch automatically without conflicts (if any).
```

Comment:
Please post below the command to delete the branch.

Reply:
I did from the github interface

Reply:
I can add a comment saying that and https://github.com/spyder-ide/spyder/branches